File: Readio/src/main/java/ninegle/Readio/publisher/repository/PublisherRepository.java
package ninegle.Readio.publisher.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import ninegle.Readio.publisher.domain.Publisher;

/**
 * Readio - PublisherRepository
 * create date:    25. 5. 9.
 * last update:    25. 5. 9.
 * author:  gigol
 * purpose:
 */
@Repository
public interface PublisherRepository extends JpaRepository<Publisher,Long> {

	Optional<Publisher> findByName(String name);
}

File: Readio/src/main/java/ninegle/Readio/publisher/repository/PublisherRepository.kt
package ninegle.Readio.publisher.repository

import ninegle.Readio.publisher.domain.Publisher
import org.springframework.data.jpa.repository.JpaRepository

/**
 * Readio - PublisherRepository
 * create date:    25. 5. 9.
 * last update:    25. 5. 9.
 * author:  gigol
 * purpose:
 */

interface PublisherRepository : JpaRepository<Publisher, Long> {
    fun findByName(name: String): Publisher?
}

